File: 1. cpp/1. basics/lecture 2/p8.cpp

#include <iostream>
using namespace std;
class solution {
public:
    solution(int n) {
      for(int i=n; i>=1; i--) 
      {
        for(int j=1; j<=(n-i); j++) 
        {
          cout<<"  ";
        }
        for(int k=1; k<=(2*i-1); k++) 
        {
          cout<<"* ";
        }
        cout<<endl;
      }
    }
};

/* pattern output:

* * * * *
  * * *
    *

*/

int main() {
  int size;
  cout<<"8. pattern height :"<<endl;
  cin>>size;

  cout<<"8. output:"<<endl;
  solution pattern(size);

}

File: 2. python/1. basics/lecture 2/p8.py
class Solution:
    def __init__(self, n):
        for i in range(n, 0, -1):
            for j in range(n - i):
                print("  ", end="") 
            
            for k in range(2 * i - 1):
                print("* ", end="") 
            
            print()

if __name__ == "__main__":
    print("8. pattern height :")
    size = int(input())

    print("8. output:")
    Solution(size)


''' shorter solution
class Solution:
    def __init__(self, n): #constructor
        for i in range(n):
            print("  " * i, end="")
            print("*" * (n - i))



if __name__ == "__main__":
    size = int(input("enter size: "))

    print("1. output:")
    Solution(size)
'''